src/Forms/GridField/GridFieldAddExistingAutocompleter.php

```diff
@@ -347,7 +348,7 @@ public function scaffoldSearchFields($dataClass)
                     // so we need to check the original setting.
                     // If the field is defined $searchable_fields = array('MyField'),
                     // then default to StartsWith filter, which makes more sense in this context.
-                    if (!$customSearchableFields || array_search($name, $customSearchableFields)) {
+                    if (!$customSearchableFields || array_search($name, $customSearchableFields) !== false) {
```

Comment:
I'm a little confused why we think `StartsWith` makes more sense in this context then `PartialMatch`.

But otherwise, the fix address a legitimate mismatch between the intention of the class and the actual behaviour.